Trivy security scan vulnerability - CVE 2025 22870

/kind bug
/area provider/ibmcloud

**What steps did you take and what happened:**
[Scan result](https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/actions/runs/13890291412/job/38860993473)
```
=== Symbol Results ===

Vulnerability #1: GO-2025-3503
    HTTP Proxy bypass using IPv6 Zone IDs in golang.org/x/net
  More info: https://pkg.go.dev/vuln/GO-2025-3503
  Module: golang.org/x/net
    Found in: golang.org/x/net@v0.33.0
    Fixed in: golang.org/x/net@v0.36.0
    Example traces found:
Error:       #1: pkg/cloud/services/cos/service.go:102:51: cos.NewService calls httpproxy.config.proxyForURL, which calls httpproxy.config.useProxy

Your code is affected by 1 vulnerability from 1 module.
This scan also found 1 vulnerability in packages you import and 3
vulnerabilities in modules you require, but your code doesn't appear to call
these vulnerabilities.
Use '-show verbose' for more details.
No vulnerabilities found.
make[1]: *** [Makefile:523: verify-govulncheck] Error 1
make[1]: Leaving directory '/home/runner/work/cluster-api-provider-ibmcloud/cluster-api-provider-ibmcloud'
Check for vulnerabilities failed! There are vulnerabilities to be fixed
make: *** [Makefile:531: verify-security] Error 1
```

```
    Found in: golang.org/x/net@v0.33.0
    Fixed in: golang.org/x/net@v0.36.0
```

- 0.36.0 version needs golang >= 1.23.x
- our release branches `release-0.9` and `release-0.10` use golang 1.22, so fixing it with go 1.22 is not possible
- As per `govulncheck` we are not using code that is affected by CVE 2025 22870

**What did you expect to happen:**
Green status for trivy security scan (no vulnerabilities)

**Anything else you would like to add:**
Solution - Ignore the reported CVE 2025 22870, similar approcah adopted in CAPI for the same.
Ref - https://github.com/kubernetes-sigs/cluster-api/pull/11976


**Environment:**

- Cluster-api version:
- Minikube/KIND version:
- Kubernetes version: (use `kubectl version`):
- OS (e.g. from `/etc/os-release`):

--- a/.trivyignore
+++ b/.trivyignore
@@ -1,0 +1,2 @@
+# According to govulncheck we are not using code that is affected by CVEs
+CVE-2025-22870

